File: 0728-self-dividing-numbers/0728-self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        answer =[]
        for i in range(left,right+1):
            flag= False
            temp =i
            original =i
            while(temp>0):
                z=temp%10
                if(z==0 or original%z !=0):
                    flag = True
                    break
                temp=temp//10
            if(flag==False):
                answer.append(i)
        return answer

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> answer = new ArrayList<>();
        
        int z;
        for(int i=left;i<=right;i++){
            boolean flag=false;
            int original=i;
            int temp =i;
            while(temp>0){
                z=temp%10;
                
                if( z==0 || original%z !=0 ){
                    flag=true;
                    break;
                }
                temp=temp/10;
            }
            if(flag==false){
                answer.add(i);
            }


        }
        return answer;
    }
}